homeassistant/components/hydrawise/sensor.py

```diff
@@ -63,15 +120,45 @@ class HydrawiseSensor(HydrawiseEntity, SensorEntity):
 
     def _update_attrs(self) -> None:
         """Update state attributes."""
-        if self.entity_description.key == "watering_time":
-            if (current_run := self.zone.scheduled_runs.current_run) is not None:
-                self._attr_native_value = int(
-                    current_run.remaining_time.total_seconds() / 60
-                )
-            else:
-                self._attr_native_value = 0
-        elif self.entity_description.key == "next_cycle":
-            if (next_run := self.zone.scheduled_runs.next_run) is not None:
-                self._attr_native_value = dt_util.as_utc(next_run.start_time)
-            else:
-                self._attr_native_value = datetime.max.replace(tzinfo=dt_util.UTC)
+        specific_getter = getattr(self, f"_get_{self.entity_description.key}")
+        value = specific_getter()
+        if value is not None:
+            self._attr_native_value = value
+
+    def _get_watering_time(self) -> int:
+        if (current_run := self.zone.scheduled_runs.current_run) is not None:
+            return int(current_run.remaining_time.total_seconds() / 60)
+        return 0
+
+    def _get_next_cycle(self) -> datetime:
+        if (next_run := self.zone.scheduled_runs.next_run) is not None:
+            return dt_util.as_utc(next_run.start_time)
+        return datetime.max.replace(tzinfo=dt_util.UTC)
+
+    def _get_daily_active_water_use(self) -> Any:
```

Comment:
I think this always returns a float?

Reply:
Let me play around with that some more. I had it like that before but I couldn't get `mypy` to play along.

Reply:
`mypy` doesn't seem to recognize `ControllerWaterUseSummary`. Maybe because it is not exported [here](https://github.com/dknowles2/pydrawise/blob/main/pydrawise/__init__.py#L13)?

Reply:
The library doesn't have a py.typed file, so mypy interprets the library types as Any practically. If the typing in your lib is on par, consider adding the py.typed file and enabling strict typing here

Reply:
I added a py.typed file and sent https://github.com/home-assistant/core/pull/116449 to bump the library.

Reply:
Fixed the return types.